src/clusterfuzz/_internal/issue_management/jira/issue_tracker_manager.py

```diff
@@ -75,16 +75,27 @@ def _transition_issue_status_if_updated(self, issue):
     if not isinstance(issue.status, jira.resources.Resource):
       self.client.transition_issue(issue.jira_issue, transition=issue.status)
 
-  def _add_watchers(self, issue):
+  def _add_watchers(self, issue, update_if_different=False):
     """Add watchers to the ticket. Jira has a separate endpoint to
     add watchers."""
 
+    # return if issue was not created yet
+    if issue.id == -1:
+      return
+
     # Get watchers from LabelStore.
-    watchers = list(issue.ccs)
+    watchers = set(issue.ccs)
+
+    if update_if_different:
+      # Only add the missing watchers
+      watchers = watchers - set(self.get_watchers(issue))
 
     # Jira weirdness, update watchers this way.
     for watcher in watchers:
-      self.client.add_watcher(issue.jira_issue, watcher)
+      try:
+        self.client.add_watcher(issue.jira_issue, watcher)
+      except Exception as e:
+        logs.error(f'Error adding watcher {watcher} to issue {issue.id}')
```

Comment:
Could you add some information from the exception on the extras field of the log.error? Like, logs.error("f'Error adding watcher {watcher} to issue {issue.id}', exception_extra=e.somefield)

Reply:
Make sense! Thanks!